terraform/031-email-service/main.tf

```diff
@@ -55,59 +57,65 @@ resource "random_id" "access_token_idsync" {
   byte_length = 16
 }
 
+
 /*
- * Create role for access to SES
+ * Create ECS role
  */
-resource "aws_iam_role" "ses" {
-  name = "ses-${var.idp_name}-${var.app_name}-${var.app_env}-${local.aws_region}"
+module "ecs_role" {
+  source = "../ecs-role"
+
+  name = "ecs-${var.idp_name}-${var.app_name}-${var.app_env}-${local.aws_region}"
+}
 
-  assume_role_policy = jsonencode({
+resource "aws_iam_role_policy" "ses" {
+  name = "ses"
+  role = module.ecs_role.role_name
```

Comment:
Previously, it looks like this `aws_iam_role_policy.ses.role` was being set to an id (`aws_iam_role.ses.id`), and now it seems to be being given a role _name_. Is this correct?

Reply:
On [`aws_iam_role`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#id), `name` and `id` are the same. I thought `name` seemed more accurate semantically.